src/main/scala/com/kubukoz/next/Spotify.scala

```diff
@@ -124,25 +128,80 @@ object Spotify {
 
     }
 
-    def build[F[_]: Concurrent: Console](sonosBaseUrl: Uri, client: Client[F]): F[Playback[F]] =
-      Console[F].println(show"Checking if Sonos API is available at $sonosBaseUrl...") *>
-        client
-          .get(sonosBaseUrl / "zones") {
-            case response if response.status.isSuccess => response.as[sonos.SonosZones].map(_.some)
-            case _                                     => none[sonos.SonosZones].pure[F]
-          }
-          .handleError(_ => None)
+    def suspend[F[_]: FlatMap](choose: F[Playback[F]]): Playback[F] = new Playback[F] {
+      def nextTrack: F[Unit] = choose.flatMap(_.nextTrack)
+      def seek(ms: Int): F[Unit] = choose.flatMap(_.seek(ms))
+    }
+
+    def build[F[_]: Concurrent: Console](sonosBaseUrl: Uri, client: Client[F]): F[Playback[F]] = {
+      def spotifyInstanceF(lastUsedRoom: RefSink[F, None.type]) = lastUsedRoom.set(None).as(spotify(client))
+
+      def sonosInstanceF(lastUsedRoom: Ref[F, Option[String]]): F[Option[Playback[F]]] = {
+
+        val fetchZones: F[Option[sonos.SonosZones]] = Console[F].println(show"Checking if Sonos API is available at $sonosBaseUrl...") *>
```

Comment:
This whole switching/printing logic is pretty fragile, some magic DSL might be cool to introduce.